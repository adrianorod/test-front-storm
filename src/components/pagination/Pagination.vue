<template>
  <div class="text-center mt-6 pagination">
    <v-btn
      :disabled="page === 1"
      @click.prevent="updatePage(1)"
      color="#D83367"
      data-test="btn-first"
      depressed
      min-height="48"
    >
      Primeiro
    </v-btn>
    <v-btn
      :disabled="page === 1"
      @click.prevent="updatePage(page - 1)"
      color="#D83367"
      depressed
      min-height="48"
    >
      Anterior
    </v-btn>

    <v-btn
      :key="item"
      @click.prevent="updatePage(item)"
      color="#D83367"
      data-test="btn-page"
      depressed
      min-height="48"
      min-width="48"
      v-for="item in pageCount"
    >
      {{ item }}
    </v-btn>

    <v-btn
      :disabled="page === pageCount"
      @click.prevent="updatePage(page + 1)"
      color="#D83367"
      data-test="btn-next"
      depressed
      min-height="48"
    >
      Próximo
    </v-btn>
    <v-btn
      :disabled="page === pageCount"
      @click.prevent="updatePage(pageCount)"
      color="#D83367"
      depressed
      min-height="48"
    >
      Último
    </v-btn>
  </div>
</template>

<script>
export default {
  name: 'Pagination',

  props: {
    page: Number,
    pageCount: Number,
  },

  methods: {
    updatePage(page) {
      this.$emit('update:page', page);
    },
  },
};
</script>

<style lang="scss">
.pagination {
  > .v-btn {
    color: #fff;
    font-weight: 300;
    letter-spacing: 0;
    margin: 0 4px;
    text-transform: capitalize;

    &--disabled,
    &.theme--light.v-btn--disabled:not(.v-btn--flat):not(.v-btn--text):not(.v-btn--outlined) {
      background-color: #FAFAFA !important;
    }
  }
}
</style>
