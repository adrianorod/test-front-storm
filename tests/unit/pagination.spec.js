import { createLocalVue, mount } from '@vue/test-utils';
import Pagination from '@/components/pagination/Pagination.vue';
import Vuetify from 'vuetify';

const localVue = createLocalVue();

describe('Pagination.vue', () => {
  let wrapper;

  beforeEach(() => {
    const vuetify = new Vuetify();

    wrapper = mount(Pagination, {
      localVue,
      vuetify,
      propsData: { page: 1, pageCount: 3 },
    });
  });

  it('should match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders page numbers correctly', () => {
    expect(wrapper.findAll('[data-test="btn-page"]')).toHaveLength(3);
  });

  it('should emit an event when an active button is clicked', () => {
    const event = jest.fn();
    const button = wrapper.find('[data-test="btn-next"]');
    wrapper.vm.$on('update:page', event);

    button.trigger('click');
    expect(event).toHaveBeenCalledTimes(1);
  });

  it('should not emit an event when an disabled button is clicked', () => {
    const event = jest.fn();
    const button = wrapper.find('[data-test="btn-first"]');
    wrapper.vm.$on('update:page', event);

    button.trigger('click');
    expect(event).toHaveBeenCalledTimes(0);
  });
});
