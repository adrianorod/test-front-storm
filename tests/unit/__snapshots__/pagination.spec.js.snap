// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination.vue should match snapshot 1`] = `
<div class="text-center mt-6 pagination"><button type="button" disabled="disabled" class="v-btn v-btn--depressed v-btn--disabled theme--light v-size--default" style="min-height: 48px;" data-test="btn-first"><span class="v-btn__content">
    Primeiro
  </span></button> <button type="button" disabled="disabled" class="v-btn v-btn--depressed v-btn--disabled theme--light v-size--default" style="min-height: 48px;"><span class="v-btn__content">
    Anterior
  </span></button> <button type="button" class="v-btn v-btn--depressed theme--light v-size--default" style="min-height: 48px; min-width: 48px; background-color: rgb(216, 51, 103); border-color: #d83367;" data-test="btn-page"><span class="v-btn__content">
    1
  </span></button><button type="button" class="v-btn v-btn--depressed theme--light v-size--default" style="min-height: 48px; min-width: 48px; background-color: rgb(216, 51, 103); border-color: #d83367;" data-test="btn-page"><span class="v-btn__content">
    2
  </span></button><button type="button" class="v-btn v-btn--depressed theme--light v-size--default" style="min-height: 48px; min-width: 48px; background-color: rgb(216, 51, 103); border-color: #d83367;" data-test="btn-page"><span class="v-btn__content">
    3
  </span></button> <button type="button" class="v-btn v-btn--depressed theme--light v-size--default" style="min-height: 48px; background-color: rgb(216, 51, 103); border-color: #d83367;" data-test="btn-next"><span class="v-btn__content">
    Próximo
  </span></button> <button type="button" class="v-btn v-btn--depressed theme--light v-size--default" style="min-height: 48px; background-color: rgb(216, 51, 103); border-color: #d83367;"><span class="v-btn__content">
    Último
  </span></button></div>
`;
